lib/web/terminal.go

```diff
@@ -874,6 +884,21 @@ func WithTerminalStreamResizeHandler(resizeC chan<- *session.TerminalParams) fun
 	}
 }
 
+func NewWStream(ws WSConn) (*WsStream, error) {
+	switch {
+	case ws == nil:
+		return nil, trace.BadParameter("required parameter ws not provided")
+	}
```

Comment:
This seems fairly ominious? Is it reasonable to always check every single argument for a nil internally in every single function? Wouldn't it make more sense to simply require that it mustn't be since this is very much an internal API. This method seems like it does basically one thing that should be the responsibility of the caller IMO.